subsys/bluetooth/host/adv.c

```diff
@@ -430,6 +430,20 @@ static inline bool ad_has_name(const struct bt_data *ad, size_t ad_len)
 	return false;
 }
 
+static inline bool ad_is_limited(const struct bt_data *ad, size_t ad_len)
+{
+	size_t i;
+
+	for (i = 0; i < ad_len; i++) {
+		if (ad[i].type == BT_DATA_FLAGS &&
```

Comment:
Perhaps we also need a length check to ensure that we can validate the data pointer (which perhaps also should be checked for non-null).